collection_requirements.rst

```diff
@@ -0,0 +1,149 @@
+
+Be sure you're subscribed to:
+* `Changes impacting Collections <https://github.com/ansible-collections/overview/issues/45>` to track changes that Collection maintainers should be aware of 
+* The Bullhorn, a newsletter for the Ansible developer community, `back issues and how to add content <https://github.com/ansible/community/issues/546>`_
+
+Why is this needed
+===================
+
+
+Repo structure
+===============
+
+galaxy.yml
+----------
+
+* tags MUST be set
+* dependencies MUST be set to ``'>=1.0.0'`` (Might have to think about dependency order when publishing)
+
+meta/runtime.yml
+----------------
+Example: `meta/runtime.yml <https://github.com/ansible-collections/collection_template/blob/main/meta/runtime.yml>`_
+* MUST define the minimum version of Ansible which this collection works with
+  * If the collection works with Ansible 2.9, then this should be set to `>=2.9.10`
+  * It's usually better to avoid adding `<2.11` as a restriction, since this for example makes it impossible to use the collection with the current ansible-base devel branch (which has version 2.11.0.dev0)
+
+Modules & Plugins
+------------------
+
+Documentation
+~~~~~~~~~~~~~~
+All module and plugin ``DOCUMENTATION`` and ``RETURN`` MUST:
+
+* Use the FQCN for ``M(...)`` See `Linking within module documentation<https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-within-module-documentation>`_
+* Use collection version numbers for `version_added` (otherwise `version_added_collection` must be provided, but this is discouraged)
+
+All module and plugin ``EXAMPLES`` MUST:
+
+* Use FQCN for module (or plugin) name.
+* for modules (or plugins) left in ansible-base use ``ansible.builtin.template``
+
+Other items:
+* You MUST Use the FQCN for ``extends_documentation_fragment:``, unless you are referring to doc_fragments from ansible-base
+
+
+Contributor Workflow
+====================
+
+Changelogs
+~~~~~~~~~~
+
+To give a consistent feel for changelogs across collections, and ensure for collections included in the ``ansible`` package we suggest you use `antsibull-changelog <https://github.com/ansible-community/antsibull-changelog>`_
+
+
+Preferred (in descending order):
+
+1. Use antsibull-changelog (preferred)
+2. Provide ``changelogs/changelog.yaml`` in the `correct format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_
+3. Provide a link to the changelog file (self-hosted) (not recommended)
+
+Please note that the porting guide is compiled from ``changelogs/changelog.yaml`` (sections ``breaking_changes``, ``major_changes``, ``deprecated_features``, ``removed_features``). So if you use option 3, you will not be able to add something to the porting guide.
+
+Versioning and deprecation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+FIXME Details here
+* Deprecation vs Collection version
+* How many Collection major versions to give
+
+Repo management
+===============
+
+Repo name
```

Comment:
```suggestion
Repo name
```
Roaming around the collections i noticed a bit strange things as parts of FQCNs like `vyos.vyos.*`. We definitely should put some recommendations here. Maybe later though

Reply:
For documentation, avoid shortened name also, if we can use "Repository"

Reply:
Yes, indeed, and with al modules name starting with also `vyos_` so you end up with `vyos.vyos.vyos_foo` ... . I tried to open issues about that on some projects, but now it's a bit too late.

Reply:
@Andersson007 Do you think we should providing some naming advice for the collections ?

Reply:
@bmillemathias sure, it could be general points without giving concrete examples how people shouldn't do :)
E.g. meaning (and we can see something similar in every book about programming)
```
- try not to use very long names but if you think it's better / necessary, feel free to do it
- choosing a name try to avoid repetitive parts in it
- ...
```
Something like that.
I definitely shoul think about it. If you feel the same and have something to suggest, it would be really great:)